.github/workflows/johnnytest4.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+output_file="CMake_Error_Log.txt"
+
+# Clean the output file first
+echo "" > $output_file
+
+# specify the directory
+directory="./.github/workflows"
+
+# loop over each file in directory
+for input_file in $directory/*.log; do
+
+  echo "File: $input_file" >> $output_file
+  echo "" >> $output_file
+  echo "Error Message:" >> $output_file
+  echo "" >> $output_file
+
+  awk '/CMake Error/{flag=1}/^$/{flag=0}flag' $input_file >> $output_file
+
+  # Load error message into a variable
+  error_message=$(awk '/CMake Error/{flag=1}/^$/{flag=0}flag' $input_file)
+  
```

Comment:
It works, but if you should do same thing twice, you may considering to avoid it if it possible.
In this case, you can store the first awk result in a variable, then echo that variable into the $output_file. That means: store the awk result into error_message variable, then echo it into the $output_file